### ISSUE TITLE
Cells made by John von Neumann with level more than 1 still have cost of upgrading like if they were level 1

### ISSUE THREAD

**Issue by Arniiiii** [ochadenas/cpudefense#201]
Cells made by John von Neumann with level more than 1 still have cost of upgrading like if they were level 1

Example:
![image](https://github.com/user-attachments/assets/58bf92fa-6bb4-469e-842b-5e930cc38070)

here's a resistor of level 2 has costs of upgrading like a lvl 1.

I see next solution:
Make cost of upgrading corresponding to the current level
and maybe create a hero that set the level of the new cells

**Comment by ochadenas** [comment]
You are right, the "value" of the chip, and depending on that the upgrading price, is (currently) not the same as if the player had bought it. The idea was that the additional chips are not just an additional source of cash (because it would be best to sell them off immediately).

I will change this so that these additional chips have the "real" value and also the real upgrading price. But in compensation, their selling price is halved.

**Comment by ochadenas** [comment]
There is (or was) another problem with von Neumann.

The additional chips are generated on stage creation, and with the option "Keep endless levels" they are part of the stage and appear at each replay. This allows to hire von Neumann, create the level, and then sell him off and spend the coins elsewhere.

I've changed this so that new chips are generated (both helpful and unwanted) each time that a stage is replayed.

**Comment by ochadenas** [comment]
changed in 1.51
